core/services/ocrcommon/transmitter_pipeline.go

```diff
@@ -0,0 +1,100 @@
+package ocrcommon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/txmgr"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+)
+
+const txObservationSource = `
+    transmit_tx [type=ethtx
+                 minConfirmations=0
+                 to="$(jobSpec.contractAddress)"
+                 from="[$(jobSpec.fromAddress)]"
+                 evmChainID="$(jobSpec.evmChainID)"
+                 data="$(jobSpec.data)"
+                 gasLimit="$(jobSpec.gasLimit)"]
+    transmit_tx
+`
+
+type pipelineTransmitter struct {
+	lgr                         logger.Logger
+	fromAddress                 common.Address
+	gasLimit                    uint32
+	effectiveTransmitterAddress common.Address
+	strategy                    txmgr.TxStrategy
+	checker                     txmgr.TransmitCheckerSpec
+	pr                          pipeline.Runner
+	spec                        job.Job
+	chainID                     string
+}
+
+// NewPipelineTransmitter creates a new eth transmitter using the job pipeline mechanism
+func NewPipelineTransmitter(
+	lgr logger.Logger,
+	fromAddress common.Address,
+	gasLimit uint32,
+	effectiveTransmitterAddress common.Address,
+	strategy txmgr.TxStrategy,
+	checker txmgr.TransmitCheckerSpec,
+	pr pipeline.Runner,
+	spec job.Job,
+	chainID string,
+) Transmitter {
+	return &pipelineTransmitter{
+		lgr:                         lgr,
+		fromAddress:                 fromAddress,
+		gasLimit:                    gasLimit,
+		effectiveTransmitterAddress: effectiveTransmitterAddress,
+		strategy:                    strategy,
+		checker:                     checker,
+		pr:                          pr,
+		spec:                        spec,
+		chainID:                     chainID,
+	}
+}
+
+func (t *pipelineTransmitter) CreateEthTransaction(ctx context.Context, toAddress common.Address, payload []byte) error {
+	vars := pipeline.NewVarsFrom(map[string]interface{}{
+		"jobSpec": map[string]interface{}{
+			"contractAddress": toAddress.String(),
+			"fromAddress":     t.fromAddress.String(),
+			"gasLimit":        t.gasLimit,
+			"evmChainID":      t.chainID,
+			"data":            payload,
+		},
+	})
+
+	run := pipeline.NewRun(pipeline.Spec{
+		ID:                t.spec.ID,
+		DotDagSource:      txObservationSource,
+		CreatedAt:         t.spec.CreatedAt,
+		MaxTaskDuration:   t.spec.MaxTaskDuration,
+		GasLimit:          &t.spec.GasLimit.Uint32,
+		ForwardingAllowed: t.spec.ForwardingAllowed,
+		JobID:             t.spec.PipelineSpecID,
+		JobName:           t.spec.Name.ValueOrZero(),
+		JobType:           t.spec.Type.String(),
+	}, vars)
```

Comment:
setting the obs source on spec might be a simpler way
https://github.com/smartcontractkit/chainlink/blob/develop/core/services/keeper/upkeep_executer.go#L246-L247

Reply:
Basically, this is the common transmitter that could be used for any kind of job. This logic should do exactly the same as the normal transmitter does so I would not delegate the observation source definition to the specific job level but would define it exactly within the pipeline transmitter.